locale.c

```diff
@@ -1345,7 +1345,18 @@ S_emulate_setlocale_i(pTHX_
              line, new_obj));
 
 #ifdef MULTIPLICITY
+
+    if (DEBUG_Lv_TEST) {
+        if (PL_cur_locale_obj != new_obj) {
+            PerlIO_printf(Perl_debug_log,
+                          "(%" LINE_Tf "): PL_cur_locale_obj"
+                          " was %p, now is %p\n",
+                          line, PL_cur_locale_obj, new_obj);
```

Comment:
I don't think `locale_t` is required to be a pointer type.

I've probably missed this elsewhere.  I think it can be assumed to be a scalar type (pointer or arithmetic type), since you can cast `0` to `locale_t`.

Reply:
Actually POSIX 2017, says 

> Upon successful completion, the newlocale() function shall return a handle

so I suppose that doesn't have to be a pointer; I've just never seen an implementation that isn't.  There are plenty of `%p` instances in blead for printing locale objects.
I see three possibilities:

1.  Continue using `%p` until and if it becomes a problem some day.
2. Add a Configure probe and a `LOCALE_Tf` define and convert to use it now
3. Just add the `LOCALE_Tf` define and set it to `%p`, convert to use it, and add a Configure probe if and when it becomes needed

Reply:
I went [by](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/locale.h.html):
```
The <locale.h> header shall define the locale_t type, representing a locale object.
```
which doesn't place any constraints on the type of `locale_t`, and being able to cast `0` to to `locale_t`.

I think 1 is fine.

Technically the argument of `%p` is required to be pointer to void (rather than any pointer) but I expect we break this rule generally.